src/main/java/org/csanchez/jenkins/plugins/kubernetes/PodTemplate.java

```diff
@@ -274,6 +277,15 @@ public void setImagePullSecrets(List<PodImagePullSecret> imagePullSecrets) {
         this.imagePullSecrets.addAll(imagePullSecrets);
     }
 
+    @DataBoundSetter
+    public void setNodeProperties(List<ToolLocationNodeProperty> nodeProperties){
+        this.nodeProperties = nodeProperties;
+    }
+
+    public List<ToolLocationNodeProperty> getNodeProperties(){
+        return nodeProperties;
```

Comment:
should return `emptyList` if nodeProperties is null ?

please set the `@Nonnull` annotation if that's the case